app/uk/gov/hmrc/taxhistory/controllers/PayAsYouEarnController.scala

```diff
@@ -17,17 +17,21 @@
 package uk.gov.hmrc.taxhistory.controllers
 
 import javax.inject.Inject
+import play.api.mvc.{Action, AnyContent, ControllerComponents}
 import uk.gov.hmrc.domain.Nino
-import uk.gov.hmrc.play.http.logging.MdcLoggingExecutionContext.fromLoggingDetails
 import uk.gov.hmrc.taxhistory.services.{EmploymentHistoryService, SaAuthService}
 import uk.gov.hmrc.time.TaxYear
 
+import scala.concurrent.ExecutionContext
+
 class PayAsYouEarnController @Inject()(employmentHistoryService: EmploymentHistoryService,
-                                       saAuthService: SaAuthService) extends TaxHistoryController {
+                                       saAuthService: SaAuthService,
+                                       val cc: ControllerComponents)(implicit val ec: ExecutionContext) extends TaxHistoryController(cc) {
 
-  def getPayAsYouEarn(nino: Nino, taxYear: TaxYear) = {
-    saAuthService.checkSaAuthorisation(nino).async { implicit request =>
+  def getPayAsYouEarn(nino: Nino, taxYear: TaxYear): Action[AnyContent] = Action.async { implicit request =>
```

Comment:
Oh just noticed we moved from an Auth action to an auth service, not fussed with either but curious around the change.

Reply:
It was basically because I couldn't get the tests to pass. The ATs were working but I kept getting NullPointerExeceptions in unit tests so I decided to try making it a service instead like the RelationshipAuthService. (Sadly that didn't fix the tests, it ended up being a sneaky mockito matching issue but I'd already changed it by then)